cvat/apps/engine/models.py

```diff
@@ -60,10 +67,36 @@ class Segment(models.Model):
     start_frame = models.IntegerField()
     stop_frame = models.IntegerField()
 
+    def get_task_name(self):
+        return self.task.name
+    get_task_name.short_description = 'Task'
+
+    def get_annotator(self):
+        return self.job_set.first().annotator
+
+    def check_user_access(self, user):
```

Comment:
We have 3 groups at the moment:
- Annotator: can view/annotate all tasks (it isn't the best idea and I see that you want to change that)
- User: potentially a person who can create tasks and delete/update own tasks as "owner" or "creator" of the task. Also can behave as an annotator. 
- Admin: can view/modify all tasks and has access to admin panel. Also can behave as an user.

I see that in your scheme there are only 2 groups: annotator and admin. Will it be good for all users of the tool?

I general your idea is very interesting but need to think how to integrate it and make all users happy. Could you please describe your proposal and together we will find a good way to implement it?